### PR TITLE
Optimize PredicateChecker for dedicated workloads

### PR DESCRIPTION
**Pull request by jkaniuk**
/cc losipiuk
/cc maciekpytel

### CODE REVIEW

File: cluster-autoscaler/simulator/predicates.go [hunk 1]
```diff
@@ -204,6 +204,27 @@ func NewTestPredicateChecker() *PredicateChecker {
 	}
 }
 
+// NewCustomTestPredicateChecker builds test version of PredicateChecker with additional predicates.
+func NewCustomTestPredicateChecker(customPredicates ...predicates.FitPredicate) *PredicateChecker {
+	newPredicates := []predicateInfo{
+		{name: "default", predicate: predicates.GeneralPredicates},
```
Comment (losipiuk) [inline cluster-autoscaler/simulator/predicates.go]: It looks kinda too arbitrary to me that we have some 2 predicates hard coded and extend it with some extra ones.

Comment (losipiuk) [reply cluster-autoscaler/simulator/predicates.go]: Is there something special about "default" and "ready" that we always want them?

Comment (jkaniuk) [reply cluster-autoscaler/simulator/predicates.go]: Sure, fixed

**Comment by losipiuk** [comment]
Seems fine to me.
/lgtm
/approve
/hold

**Comment by losipiuk** [comment]
Please unhold as soon as you want to merge

**Pull request merged**